### ISSUE TITLE
Fix false negatives for decorators

### ISSUE THREAD

**Issue by abdonrd** [stylelint/stylelint#4761]
> Clearly describe the bug

Stylelint doesn't work with decorators

> What code is needed to reproduce the bug?

Create this file: `page-home.ts`

```typescript
import { LitElement, html, css, customElement } from 'lit-element';

@customElement('page-home')
export class PageHome extends LitElement {
  static styles = css`
    :host {
      display: block;
      display: block;
      padding: 1rem;
    }
  `;

  render() {
    return html`
      <section>
        <h1>Home</h1>
      </section>
    `;
  }
}
```

Note the duplicated `display: block;`.

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": [
    "stylelint-config-standard"
  ],
  "rules": {
    "no-eol-whitespace": null,
    "no-missing-end-of-source-newline": null,
    "selector-type-no-unknown": [true, { "ignore": ["custom-elements"] }]
  }
}
```

> Which version of stylelint are you using?

`v13.3.3`

In previous versions it was working fine.

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "src/**/*.ts"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Template literals.

> What did you expect to happen?

Detect the errors.

> What actually happened (e.g. what warnings or errors did you get)?

Don't detect the errors.

But if I comment the decorator, it works! 😮 

```typescript
import { LitElement, html, css, customElement } from 'lit-element';

// @customElement('page-home')
export class PageHome extends LitElement {
  static styles = css`
    :host {
      display: block;
      display: block;
      padding: 1rem;
    }
  `;

  render() {
    return html`
      <section>
        <h1>Home</h1>
      </section>
    `;
  }
}
```

```shell
src/page-about.ts
 8:7  ✖  Unexpected duplicate "block"   declaration-block-no-duplicate-properties
```

**Comment by jeddy3** [comment]
@abdonrd Thanks for the report and for using the template.

I can [reproduce this using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcID0SACYCuBnALhAtgKIA2M+MAdjgBQDkAGrQJQA6FMAHgA4QBOOAAjDEAhliwD6AzjkpQJATQHA2AgbhE4AlmHU4AnqQkBeIeIAGqtQPgALCLmVXrAqFqxdR++AIBGxCDAAawBuZ2s3Dy8ff0DQ8IBfK3MwigSQABoQLH0qEQ4EcHEAWi0KYoArLEzwCAoAMy0Ac0KVCjUWEF4MI06fNpdO2GERXk0tOuLY4OKKCGKoDE8dTRhirl4ILhh+LRgsPoEAbXC1HG6YDNOndpdrTua53hhDk9u7+9qKLBhMbQA3NaLZZgVZYYoAdy0OFsCy09XqO0oOGK-xExAw+061zUAF1rkl3gJ8bdCekEkA) with:

```js
//@customElement('X')
export class X extends Y {
  static styles = css`
    :host {
      display: block;
      display: block;
    }
  `;
}
```

It's likely a problem with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a growing list of the problems with the syntax that we need help to fix.

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time. We recently forked it into the stylelint organisation.

**Comment by abdonrd** [comment]
@jeddy3 any workaround in the meantime? Because I have tried to fix the `v13.3.0` and the `v13.0.0`, but it still doesn't work. And a while ago it did.

**Comment by jeddy3** [comment]
> any workaround in the meantime?

Not that I'm aware of.

I tried passing a [year-old version of the parser](https://www.npmjs.com/package/postcss-jsx) into stylelint using the [`customSyntax` option](https://stylelint.io/user-guide/usage/options#customsyntax), but that didn't work either.

The person who originally wrote the CSS-in-JS parser is no longer active, so we recently [forked it into stylelint organisation](https://github.com/stylelint/postcss-css-in-js) to patch a couple of bugs. Unfortunately, there's been little time to do more.

The parser is looking for someone, a group of people or a company to champion it, i.e. fix bugs, review pull requests and publish releases.

**Comment by joemaffei** [comment]
Not sure if it's related to this issue, but I'm getting a "Leading decorators must be attached to a class declaration" when linting a css file that contains font-face declarations.

**Comment by aress31** [comment]
Any update on the issue?

**Comment by simonwinter** [comment]
> Not sure if it's related to this issue, but I'm getting a "Leading decorators must be attached to a class declaration" when linting a css file that contains font-face declarations.

Same for me ... in fact, I get it with `@define-mixin`& `@charset` decorators as well.

**Comment by abdonrd** [comment]
Any news here? Thanks in advance!

**Comment by aress31** [comment]
Should be a top priority to have this project works with TS.

**Comment by hudochenkov** [comment]
@aress31 all help is welcome.

**Comment by AuthorProxy** [comment]
up

**Comment by dvakatsiienko** [comment]
+

**Comment by abdonrd** [comment]
Could someone guide us a bit?
It's a shame it's been open for so long.
Thanks in advance!

**Comment by abdonrd** [comment]
Thanks @43081j! ❤️

I just tried with this here (If I understood well):
https://github.com/IBM/pwa-lit-template/compare/stylelint-decorators

But it seems like it's not enough.

**Comment by 43081j** [comment]
aha it appears postcss-css-in-js hard codes it!

https://github.com/stylelint/postcss-css-in-js/blob/3e8bf747fd6faac3f5cb6e5418419190ccfe6154/extract.js#L97

**Comment by abdonrd** [comment]
@43081j thanks again! PR created! https://github.com/stylelint/postcss-css-in-js/pull/161

**Comment by melink14** [comment]
Looking at the disucssion in the PR it seems a workaround before that is merged is to install the babel module yourself:
`npm install @babel/plugin-proposal-decorators -D`

which caused stylelint to output the same errors as when I commented out the decorator. (Of course, in my case (LitElement) I could also use alternate syntax to avoid decorators altogether as another workaround)

**Comment by furdzik** [comment]
Hi,

I was wondering if something will be done with this.
I like using CSS-in-JS, but styles quality is important to me. 
It would be great if it would work.

If I can help to move this forward, just let me know.

**Comment by ybiquitous** [comment]
@furdzik We are discussing the deprecation for the support for CSS-in-JS.
Please see https://github.com/stylelint/postcss-css-in-js/issues/225

**Comment by jeddy3** [comment]
@furdzik Just to be clear, we only intend to deprecate the `@stylelint/css-in-js` package. Stylelint can be (and I presume always will be) extendable via the [`customSyntax` option](https://stylelint.io/user-guide/usage/options#customsyntax) to support all sorts of containers for styles, CSS-in-JS included.

I don't know what particular CSS-in-JS library your using, but our recommendation is now to [write a custom syntax](https://stylelint.io/developer-guide/syntaxes) _specifically for that library_.

If you do write a custom syntax for your library of choice, please share a link to it in thread that @ybiquitous mentioned so that others can find and benefit from it.

**Comment by puku0x** [comment]
> aha it appears postcss-css-in-js hard codes it!
>
>https://github.com/stylelint/postcss-css-in-js/blob/3e8bf747fd6faac3f5cb6e5418419190ccfe6154/extract.js#L97
>
>edit: it should be overriding that when you have a babelrc though, ill dig a bit

I'm happy to know I'm not the last one facing the problem😂
https://github.com/stylelint/postcss-css-in-js/pull/237 may be another solution in terms of it overrides the hard-coded plugin for decorators.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
